CMakeLists.txt

```diff
@@ -21,11 +21,14 @@ endif()
 message(STATUS "CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
 message(STATUS "CMAKE_SYSTEM_NAME: ${CMAKE_SYSTEM_NAME}")
 
+if(NOT DEFINED EXTRA_FLAGS)
+  set(EXTRA_FLAGS "-march=haswell" CACHE STRING
```

Comment:
Probably the default should be `native` and the users should decide whether to use a specific march value for their environment ?!